tests/test_tokenization_common.py

```diff
@@ -3166,13 +3166,42 @@ def test_training_new_tokenizer(self):
         self.assertEqual(len(inputs["input_ids"]), 2)
 
         # Test with a special tokens map
-        if tokenizer._cls_token is not None:
-            new_tokenizer = tokenizer.train_new_from_iterator(
-                SMALL_TRAINING_CORPUS, 100, special_tokens_map={tokenizer.cls_token: "<cls>"}
-            )
-            cls_id = new_tokenizer.get_vocab()["<cls>"]
-            self.assertEqual(new_tokenizer.cls_token, "<cls>")
-            self.assertEqual(new_tokenizer.cls_token_id, cls_id)
+        special_tokens_list = [
+            "bos_token",
+            "eos_token",
+            "unk_token",
+            "sep_token",
+            "pad_token",
+            "cls_token",
+            "mask_token",
+        ]
+        # Create a new mapping from the special tokens defined in the original tokenizer
+        special_tokens_map = {}
+        for spe_token_attr in special_tokens_list:
+            spe_token = getattr(tokenizer, spe_token_attr)
```

Comment:
This should actually check if there are in the signature of the tokenizer fast, because otherwise they are not updated :see_no_evil:

Reply:
Indeed! 

But I just thought about it, but it means that this feature will not work for tokenizers that are only PreTrainedTokenizerFast (such as `neuropark/sahajBERT`) ?

Reply:
Are you setting some specific token when instantiating the `PreTrainedTokenizerFast`?

Reply:
Yes, I used that for sahajBERT tokenizer :slightly_smiling_face: . Just in case, I put [here](https://huggingface.co/neuropark/sahajBERT/blob/main/tokenizer_config.json) the link to sahajBERT tokenizer_config.json.

Reply:
Mmm, will try to adjust the main PR then :-)